### PR TITLE
add solution to task 7

### PR DESCRIPTION
**Pull request by adunners**

### CODE REVIEW

**Comment by adunners** [comment]
I've realised I've forgotten to up date the endpoints.json file, I will do that before my next push

File: __tests__/app.test.js [hunk 1]
```diff
@@ -72,79 +69,165 @@ describe("app", () => {
...
+          .post("/api/articles/1/comments")
+          .send({
+            username: "butter_bridge",
+          })
+          .expect(400)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+          });
+      });
+      test("POST 400: should return with an error if an invalid username is given", () => {
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:183]: I would suggest that an invalid username should be a 404 in this case, as it's a potentially valid username but just isn't found on the users table.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -72,79 +69,165 @@ describe("app", () => {
...
+          .send({
+            username: "butter_bridge",
+            body: 500,
+          })
+          .expect(400)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+          });
+      });
+      test("POST 400: should return an error if an invalid file path is given", () => {
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:207]: We can be a bit more specific with the description here, it's actually that the article_id is invalid in this case.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -72,79 +69,165 @@ describe("app", () => {
...
+          .expect(404)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Not Found");
+          });
+      });
+    });
   });
-});
\ No newline at end of file
+});
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:233]: Really nice testing overall, just a couple of tweaks needed 👍

File: models/topics-model.js [hunk 4]
```diff
@@ -39,35 +39,75 @@ exports.fetchArticleById = (id) => {
...
+   (body, article_id, author)
+   VALUES
+   ($1, $2, $3)
+   RETURNING *
+  `,
+        [body, id, username]
+      );
+    })
+    .then((addedComment) => {
+      addedComment.rows[0].created_at = String(addedComment.rows[0].created_at);
```
Comment (nc-richard-neat) [inline models/topics-model.js:109]: Lets allow the database to dictate how the date comes back on the response. When we're asserting against it we can deal with it how it comes, it's probably fine to expect.any(String) for the data as the format is tricky to predict.

**Pull request merged**